modules/git/git.go

```diff
@@ -206,6 +206,12 @@ func Init(ctx context.Context) error {
 		SupportProcReceive = false
 	}
 
+	if CheckGitVersionAtLeast("2.36") == nil {
+		if err := checkAndSetConfig("safe.directory", "*", true); err != nil {
```

Comment:
I think we should have a comment here, explaining why this is chosen and the rationale of it being safe.

Otherwise I would already see the comments/issues/questions pouring in. of why we're doing this and why we're deliberately making Gitea insecure.

Reply:
It looks like a windows only issue - so can we only set this on windows?

Reply:
It is described for windows but is not windows specific:

```
/tmp/a/b$ strace -e stat git log
stat("/usr/share/locale", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat("/tmp/a/b", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
stat("/tmp/a/b/.git", 0x7ffe70098e60)   = -1 ENOENT (No such file or directory)
stat("/tmp/a", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
stat("/tmp/a/.git", 0x7ffe70098e60)     = -1 ENOENT (No such file or directory)
stat("/tmp", {st_mode=S_IFDIR|S_ISVTX|0777, st_size=139264, ...}) = 0
stat("/tmp/.git", 0x7ffe70098e60)       = -1 ENOENT (No such file or directory)
stat("/", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat("/.git", 0x7ffe70098e60)           = -1 ENOENT (No such file or directory)
fatal: not a git repository (or any of the parent directories): .git
+++ exited with 128 +++
```